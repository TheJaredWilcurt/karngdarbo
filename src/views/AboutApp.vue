<template>
  <div class="about-app">
    {{ appTitle }} is a Cross-Platform Git client made by
    <strong>
      <a
        href="#"
        title="http://TheJaredWilcurt.com"
        @click.prevent="openExternal"
        ref="testExternalLink"
      >
        The Jared Wilcurt
      </a>
    </strong>.

    <ul>
      <li><strong>{{ appTitle }}</strong> (v{{ versions.app }})</li>
      <li><strong>Git</strong> (v{{ versions.git }})</li>
      <li><strong>NW.js</strong> (v{{ versions.nw }})
        <ul>
          <li><strong>Chromium</strong> (v{{ versions.chromium }})</li>
          <li><strong>Node.js</strong> (v{{ versions.node }})</li>
        </ul>
      </li>
      <li><strong>Vue.js</strong> (v{{ versions.vue | clean }})</li>
    </ul>
  </div>
</template>

<script>
const exec = nw.require('child_process').exec;

export default {
  name: 'AppSettings',
  data: function () {
    return {
      appTitle: 'Karng Darbo',
      versions: {
        app: nw.App.manifest.version,
        chromium: nw.process.versions.chromium,
        git: '',
        node: nw.process.versions.node,
        nw: nw.process.versions.nw,
        vue: nw.App.manifest.dependencies.vue
      }
    };
  },
  methods: {
    getGitVersion: function () {
      exec('git --version', (err, data) => {
        if (err) {
          this.$store.commit('setAppError', String(err));
        }
        this.versions.git = data.replace('git version ', '').trim();
      });
    },
    openExternal: function (evt) {
      let url = evt.target.title;
      nw.Shell.openExternal(url);
    }
  },
  filters: {
    clean: function (val) {
      return val.replace('^', '').replace('~', '').trim();
    }
  },
  created: function () {
    this.getGitVersion();
  }
};
</script>
