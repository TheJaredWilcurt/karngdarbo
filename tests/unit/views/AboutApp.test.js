import { shallowMount } from '@vue/test-utils';

import AboutApp from '@/views/AboutApp.vue';

describe('AboutApp.vue', () => {
  describe('Created', () => {
    test('Default snapshot', () => {
      const wrapper = shallowMount(AboutApp);

      expect(wrapper.html())
        .toMatchSnapshot();
    });
  });
});
