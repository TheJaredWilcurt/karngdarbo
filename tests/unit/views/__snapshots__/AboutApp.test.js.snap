// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutApp.vue Created Default snapshot 1`] = `
<div class="about-app">
  Karng Darbo is a Cross-Platform Git client made by
  <strong>
    <a href="#" title="http://TheJaredWilcurt.com">The Jared Wilcurt</a>
  </strong>.

  <ul>
    <li>
      <strong>Karng Darbo</strong> (v0.0.0)
    </li>
    <li>
      <strong>Git</strong> (v2.12.2.windows.2)
    </li>
    <li>
      <strong>NW.js</strong> (v0.0.0)
      <ul>
        <li>
          <strong>Chromium</strong> (v0.0.0)
        </li>
        <li>
          <strong>Node.js</strong> (v0.0.0)
        </li>
      </ul>
    </li>
    <li>
      <strong>Vue.js</strong> (v0.0.0)
    </li>
  </ul>
</div>
`;
